File: invesdwin-scripting-parent/invesdwin-scripting-runtime-scala/src/test/java/de/invesdwin/context/scala/tests/callback/ParametersAndReturnsTestByte.scala
System.out.println("getByte")
if(getByte != null)
	throw new Exception("getByte already defined!")
val getByte: java.lang.Byte = callback("getByte")
val getByteType = getByte.getClass()
System.out.println(getByteType)
System.out.println(getByte)
if(getByteType != classOf[java.lang.Byte])
	throw new Exception("getByte not Byte!")
callback("setByte",getByte)

System.out.println("getByteVector")
if(getByteVector != null)
	throw new Exception("getByteVector already defined!")
val getByteVector: Array[Byte] = callback("getByteVector")
val getByteVectorType = getByteVector(0).getClass()
System.out.println(getByteVectorType)
System.out.println(getByteVector)
if(getByteVectorType != classOf[Byte])
	throw new Exception("getByteVector not Byte!")
callback("setByteVector",getByteVector)

System.out.println("getByteVectorAsList")
if(getByteVectorAsList != null)
	throw new Exception("getByteVectorAsList already defined!")
val getByteVectorAsList: Array[Byte] = callback("getByteVectorAsList")
val getByteVectorAsListType = getByteVectorAsList(0).getClass()
System.out.println(getByteVectorAsListType)
System.out.println(getByteVectorAsList)
if(getByteVectorAsListType != classOf[Byte])
	throw new Exception("getByteVectorAsList not Byte!")
callback("setByteVectorAsList",getByteVectorAsList)

System.out.println("getByteMatrix")
if(getByteMatrix != null)
	throw new Exception("getByteMatrix already defined!")
val getByteMatrix: Array[Array[Byte]] = callback("getByteMatrix")
val getByteMatrixType = getByteMatrix(0)(0).getClass()
System.out.println(getByteMatrixType)
System.out.println(getByteMatrix)
if(getByteMatrixType != classOf[Byte])
	throw new Exception("getByteMatrix not Byte!")
callback("setByteMatrix",getByteMatrix)

System.out.println("getByteMatrixAsList")
if(getByteMatrixAsList != null)
	throw new Exception("getByteMatrixAsList already defined!")
val getByteMatrixAsList: Array[Array[Byte]] = callback("getByteMatrixAsList")
val getByteMatrixAsListType = getByteMatrixAsList(0)(0).getClass()
System.out.println(getByteMatrixAsListType)
System.out.println(getByteMatrixAsList)
if(getByteMatrixAsListType != classOf[Byte])
	throw new Exception("getByteMatrixAsList not Byte!")
callback("setByteMatrixAsList",getByteMatrixAsList)

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/callback/ParametersAndReturnsTestByte.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getByte")
if(getByte ~= nil) then
	error("getByte already defined!")
end
getByte = callback("getByte")
getByteType = type(getByte)
print(getByteType)
print(getByte)
if(getByteType ~= "number") then
	error("getByte not Byte!")
end
callback("setByte", getByte)

print("getByteVector")
if(getByteVector ~= nil) then
	error("getByteVector already defined!")
end
getByteVector = callback("getByteVector")
getByteVectorType = type(getByteVector[1])
print(getByteVectorType)
print(arraysToString(getByteVector))
if(getByteVectorType ~= "number") then
	error("getByteVector not Byte!")
end
callback("setByteVector", getByteVector)

print("getByteVectorAsList")
if(getByteVectorAsList ~= nil) then
	error("getByteVectorAsList already defined!")
end
getByteVectorAsList = callback("getByteVectorAsList")
getByteVectorAsListType = type(getByteVectorAsList[1])
print(getByteVectorAsListType)
print(arraysToString(getByteVectorAsList))
if(getByteVectorAsListType ~= "number") then
	error("getByteVectorAsList not Byte!")
end
callback("setByteVectorAsList", getByteVectorAsList)

print("getByteMatrix")
if(getByteMatrix ~= nil) then
	error("getByteMatrix already defined!")
end
getByteMatrix = callback("getByteMatrix")
getByteMatrixType = type(getByteMatrix[1][1])
print(getByteMatrixType)
print(arraysToString(getByteMatrix))
if(getByteMatrixType ~= "number") then
	error("getByteMatrix not Byte!")
end
callback("setByteMatrix", getByteMatrix)

print("getByteMatrixAsList")
if(getByteMatrixAsList ~= nil) then
	error("getByteMatrixAsList already defined!")
end
getByteMatrixAsList = callback("getByteMatrixAsList")
getByteMatrixAsListType = type(getByteMatrixAsList[1][1])
print(getByteMatrixAsListType)
print(arraysToString(getByteMatrixAsList))
if(getByteMatrixAsListType ~= "number") then
	error("getByteMatrixAsList not Byte!")
end
callback("setByteMatrixAsList", getByteMatrixAsList)
